sklearn/notebooks/hp_tuning_auto_mpg/auto_mpg_hp_tuning/train.py

```diff
@@ -0,0 +1,179 @@
+# Download the data
+blob.download_to_filename('auto-mpg.data')
+# [END mle_sklearn_hp_tuning_download_data]
+
+
+# ---------------------------------------
+# This is where your model code would go. Below is an example model using the auto mpg dataset.
+# ---------------------------------------
+# [START mle_sklearn_hp_tuning_define_and_load_data]
+# Define the format of your input data including unused columns
+# (These are the columns from the auto-mpg data files)
+COLUMNS = (
+    'mpg',
+    'cylinders',
+    'displacement',
+    'horsepower',
+    'weight',
+    'acceleration',
+    'model-year',
+    'origin',
+    'car-name'
+)
+
+# Load the training auto mpg dataset
+with open('./auto-mpg.data', 'r') as train_data:
+    raw_training_data = pd.read_csv(train_data, header=None, names=COLUMNS, delim_whitespace=True)
+
+# Remove the column we are trying to predict ('mpg') from our features list
+# Convert the Dataframe to a lists of lists
+features = raw_training_data.drop('mpg', axis=1).drop('car-name', axis=1).values.tolist()
+
+# Create our training labels list, convert the Dataframe to a lists of lists
+labels = raw_training_data['mpg'].values.tolist()
+
+train_features, test_features, train_labels, test_labels = train_test_split(features, labels, test_size=0.15)
+# [END mle_sklearn_hp_tuning_define_and_load_data]
+
+# ---------------------------------------
+# 3. Use the value passed in those arguments to set the corresponding hyperparameters
+#    in your application's scikit-learn code.
+# ---------------------------------------
+# [START mle_sklearn_hp_tuning_create_model]
+# Create the regressor, here we will use a Lasso Regressor to demonstrate the use of HP Tuning.
+# Here is where we set the variables used during HP Tuning from
+# the parameters passed into the python script
+regressor = Lasso(
+    alpha=args.alpha,
+    max_iter=args.max_iter,
+    tol=args.tol,
+    selection=args.selection)
+
+# Transform the features and fit them to the regressor
+regressor.fit(train_features, train_labels)
+# [END mle_sklearn_hp_tuning_create_model]
+
+# ---------------------------------------
+# 4. Report the mean accuracy as hyperparameter tuning objective metric.
+# ---------------------------------------
+# [START mle_sklearn_hp_tuning_metrics]
+# Calculate the mean accuracy on the given test data and labels.
+score = regressor.score(test_features, test_labels)
+
+# The default name of the metric is training/hptuning/metric. 
+# We recommend that you assign a custom name. The only functional difference is that 
+# if you use a custom name, you must set the hyperparameterMetricTag value in the 
+# HyperparameterSpec object in your job request to match your chosen name.
+# https://cloud.google.com/ml-engine/reference/rest/v1/projects.jobs#HyperparameterSpec
+hpt = hypertune.HyperTune()
+hpt.report_hyperparameter_tuning_metric(
+    hyperparameter_metric_tag='my_metric_tag',
+    metric_value=score,
+    global_step=1000)
```

Comment:
I should add a comment to clarify. 
The global step here, (I believe), is simply at what point to report information back to CMLE for how well the model is doing. 
The reason I chose 1000 here, is just so that each model has the same step. (its default is 0)